diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -390,9 +390,11 @@ namespace diff_drive_controller
     }
 
     // Publish odometry message
-    if (last_state_publish_time_ + publish_period_ < time)
+    const ros::Duration half_period(0.5 * period.toSec());
```

Comment:
Pedantic, may as well calculate this after reading ros::param to save a few cycles, instead of calculating each time.